network/benchmarks/netperf/lib/utilslib.go

```diff
@@ -290,18 +293,37 @@ func executeTests(c *kubernetes.Clientset, testParams TestParams, primaryNode, s
 	return results, nil
 }
 
-func getOrchestratorPodName(c *kubernetes.Clientset, testNamespace string) string {
+func getOrchestratorPodName(c *kubernetes.Clientset, testNamespace string) (string, error) {
 	for {
 		fmt.Println("Waiting for orchestrator pod creation")
-		time.Sleep(60 * time.Second)
+		time.Sleep(5 * time.Second)
 		pods, err := c.CoreV1().Pods(testNamespace).List(context.Background(), everythingSelector)
 		if err != nil {
 			fmt.Println("Failed to fetch pods - waiting for pod creation", err)
 			continue
 		}
 		for _, pod := range pods.Items {
 			if strings.Contains(pod.GetName(), "netperf-orch-") {
-				return pod.GetName()
+				isPodFailed := pod.Status.Phase == api.PodFailed
+				if isPodFailed {
+					podEvent, err := c.CoreV1().Events(testNamespace).List(context.Background(), metav1.ListOptions{
+						FieldSelector: fmt.Sprintf("involvedObject.name=%s,involvedObject.kind=Pod", pod.GetName()),
+					})
+					if err != nil {
+						return "", fmt.Errorf("failed to start orchestrador pod and failed to get events for pod %s: %v", pod.GetName(), err)
+					}
+					fmt.Println("Orchestrator pod failed to start, listing pod events")
+					for _, event := range podEvent.Items {
+						fmt.Printf("Event: %s %s %s\n", event.Reason, event.Message, event.LastTimestamp)
+					}
+					return "", fmt.Errorf("orchestrator pod failed to start: %v", podEvent.Items)
+				}
+				isPodRunning := pod.Status.Phase == api.PodRunning
+				if !isPodRunning {
+					fmt.Println("Orchestrator pod is not running yet")
```

Comment:
what if pod is not running forever?

Reply:
Can we add a timeout to in case pods can't get scheduled due to node issue or wrong setup?

Reply:
Make sense, will do!